wallet/src/de/schildbach/wallet/ui/DashPayUserActivityViewModel.kt

```diff
@@ -17,91 +17,124 @@
 import de.schildbach.wallet.data.UsernameSearchResult
+import de.schildbach.wallet.data.UsernameSortOrderBy
 import de.schildbach.wallet.database.entity.DashPayProfile
 import de.schildbach.wallet.livedata.Resource
 import de.schildbach.wallet.service.DashSystemService
 import de.schildbach.wallet.service.platform.PlatformSyncService
-import de.schildbach.wallet.ui.dashpay.NotificationsForUserLiveData
 import de.schildbach.wallet.ui.dashpay.PlatformRepo
-import de.schildbach.wallet.ui.dashpay.TransactionsLiveData
 import de.schildbach.wallet.ui.dashpay.work.SendContactRequestOperation
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.dash.wallet.common.WalletDataProvider
 import org.dash.wallet.common.services.analytics.AnalyticsService
+import org.dashj.platform.dpp.identifier.Identifier
 import org.slf4j.LoggerFactory
-import java.lang.Exception
 import javax.inject.Inject
 
 @HiltViewModel
 class DashPayUserActivityViewModel @Inject constructor(
-    application: Application,
+    @ApplicationContext private val context: Context,
     val platformSyncService: PlatformSyncService,
     private val analytics: AnalyticsService,
     val platformRepo: PlatformRepo,
-    private val dashSystemService: DashSystemService
-) : AndroidViewModel(application) {
+    private val dashSystemService: DashSystemService,
+    private val walletData: WalletDataProvider
+) : ViewModel() {
 
     companion object {
-        val log = LoggerFactory.getLogger(DashPayUserActivityViewModel::class.java)
+        private val log = LoggerFactory.getLogger(DashPayUserActivityViewModel::class.java)
     }
 
-    //private val platformRepo = PlatformRepo.getInstance()
-    private val walletApplication = application as WalletApplication
-    val transactionsLiveData = TransactionsLiveData()
+    private var contactRequestStatusJob: Job? = null
 
-    lateinit var userData: UsernameSearchResult
+    private val _userData = MutableStateFlow<UsernameSearchResult?>(null)
+    val userData: StateFlow<UsernameSearchResult?>
+        get() = _userData.asStateFlow()
+
+    private val _sendContactRequestState = MutableStateFlow<Resource<Pair<String, String>>?>(null)
+    val sendContactRequestState: StateFlow<Resource<Pair<String, String>>?>
+        get() = _sendContactRequestState.asStateFlow()
+
+    private val _notifications = MutableStateFlow<List<NotificationItem>>(listOf())
+    val notifications: StateFlow<List<NotificationItem>>
+        get() = _notifications.asStateFlow()
```

Comment:
Removed all the complicated LiveData flows and exposed 3 simple states for contact data, notification, and request state. LiveDatas were often used for a single run by invoking a function and emitting the result, which is a bit convoluted and hard to follow.